### PR TITLE
Adapation of aggregator code from master thesis by Claudio Lex (#93)

### LINKED ISSUE #94: Implementation of aggregator
### Additional information

Currently, the aggregator is included in the config file but not yet implemented. For the implementation it needs to be checked how the config needs to look like to actually implement it, it needs to be considered in the scenario creation and execution. And of course the analysis once this exists.

### PR DESCRIPTION
**Pull request by TUM-Doepfert**
## TL;DR
This PR adapts the implementation of the aggregator by Claudio Lex in his master thesis. The code was adjusted to only implement the functionality but nothing thesis specific.

---

## Linked Issues
Closes #94 

## Description
Please include a summary of the change and which issue is fixed. Include the context and motivations behind your changes. If needed, add screenshots or videos.

## How to test
Describe how the PR should be tested (ideally with an enumerated list)

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist:
- [ ] My PR is concise and addresses only one bug/feature (otherwise split PR into two)
- [ ] My PR has a concise and descriptive title
- [ ] My code follows the code style of this project.
- [ ] I have performed a self-review of my own code and PR and have added comments in the PR code to help the reviewer.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.

## Additional Notes
Add any other notes or comments here.

[How to make a good PR (Video)](https://www.youtube.com/watch?v=_HedItVFr5M)

### CODE REVIEW

**Comment by TUM-Doepfert** [comment]
PR is from the wrong branch.
